File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function(s) {
    let res = 0
    let count = 0
    for (const c of s) {
        if (c === 'L') {
            count++
        } else {
            count--
        }
        if (count === 0) res++
    }
    return res
};

File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        count = ans = 0
        for c in s:
            if c == "L":
                count += 1
            else:
                count -= 1
            if count == 0:
                ans += 1
        return ans